internal/pod/kmodloader.go

```diff
@@ -0,0 +1,551 @@
+package pod
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/config"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/meta"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/worker"
+	"github.com/mitchellh/hashstructure/v2"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubectl/pkg/cmd/util/podcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+)
+
+//go:generate mockgen -source=kmodloader.go -package=pod -destination=mock_kmodloader.go KmodLoader
+
+type KmodLoader interface {
+	CreateLoaderPod(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleSpec) error
+	CreateUnloaderPod(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleStatus) error
+	DeletePod(ctx context.Context, pod *v1.Pod) error
+	GetWorkerPod(ctx context.Context, podName, namespace string) (*v1.Pod, error)
+	ListWorkerPodsOnNode(ctx context.Context, nodeName string) ([]v1.Pod, error)
+	LoaderPodTemplate(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleSpec) (*v1.Pod, error)
+	UnloaderPodTemplate(ctx context.Context, nmc client.Object, nms *kmmv1beta1.NodeModuleStatus) (*v1.Pod, error)
+}
+
+const (
+	WorkerActionLoad           = "Load"
```

Comment:
can we hide those exported variables, making them private and defining Getter/Setter interface functions if needed?

Reply:
Do you mean all the exported variables in this file?
```
        WorkerActionLoad = "Load"
	WorkerActionUnload         = "Unload"
	NodeModulesConfigFinalizer = "kmm.node.kubernetes.io/nodemodulesconfig-reconciler"
	ActionLabelKey             = "kmm.node.kubernetes.io/worker-action"
	ConfigAnnotationKey        = "kmm.node.kubernetes.io/worker-config"
	WorkerContainerName        = "worker"
	HashAnnotationKey          = "kmm.node.kubernetes.io/worker-hash"
```

Reply:
I will investigate possible options in a future PR. Let's keep this one a simple as possible (just moving code from one place to another).